cmd/builder/internal/builder/main.go

```diff
@@ -139,21 +179,37 @@ func GetModules(cfg Config) error {
 		return fmt.Errorf("failed to update go.mod: %w", err)
 	}
 
-	cfg.Logger.Info("Getting go modules")
-	// basic retry if error from go mod command (in case of transient network error). This could be improved
-	// retry 3 times with 5 second spacing interval
-	retries := 3
-	failReason := "unknown"
-	for i := 1; i <= retries; i++ {
-		if err := runGoCommand(cfg, "mod", "download"); err != nil {
-			failReason = err.Error()
-			cfg.Logger.Info("Failed modules download", zap.String("retry", fmt.Sprintf("%d/%d", i, retries)))
-			time.Sleep(5 * time.Second)
+	if cfg.SkipStrictVersioning {
+		return downloadModules(cfg)
+	}
```

Comment:
I am confused about this, shouldn't this be the opposite, i.e.:

```suggestion
	if !cfg.SkipStrictVersioning {
		return downloadModules(cfg)
	}
```

?

Reply:
the logic in `downloadModules()` was already a part of the `GetModules()` function.  Since we do strict checking before this logic, I put it into its own function that can be called while returning.

Reply:
Ah, right, sorry I got confused but the code is alright :)